server/server.py

```diff
@@ -1,21 +1,38 @@
 from flask import Flask, request, jsonify
 from newsapi import NewsApiClient
 
+import os
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
+
 #Init Flask App
 app = Flask(__name__)
 
 #Init News Api
-newsapi = NewsApiClient(api_key = '31673bb1bed245e8b253b67389155bb2')
-
-
+newsapi = NewsApiClient(api_key=os.getenv("NEWS_API_KEY"))
 
+#API endpoint for handling search request based on user input
 @app.route("/search", methods = ["GET", "POST"])
 def search():
     if request.method == "GET":
         searched_term = request.args.get("q")
-        searched_articles = newsapi.get_everything(q=searched_term)['articles']
-    
-    return jsonify(searched_articles)
+
+        try:
+            response = newsapi.get_everything(q=searched_term, language='en', sort_by='relevancy', page = 1)
+            # Check the response status from NewsAPI
+            if response['status'] == 'ok':
+                searched_articles = response['articles']
+                return jsonify(searched_articles)
+            elif response['status'] == 'error':
```

Comment:
```suggestion
            else:  # status = 'error'
```
status chỉ có 2 TH là ok và error nên k cần elif

để dev cũng hiểu là nó chỉ có 2 TH thì comment như trên là đc

Reply:
Dạ em đã sửa lại để dùng elif cho rõ hơn rồi ạ.

Reply:
resolved